engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala

```diff
@@ -194,15 +201,31 @@ class PeriodicProcessService(delegateDeploymentManager: DeploymentManager,
   }
 
   /**
-    * Returns latest deployment. It can be in any status (consult [[PeriodicProcessDeploymentStatus]]).
-    *
-    * For multiple schedules only the oldest one is returned.
-    */
+   * Returns latest deployment. It can be in any status (consult [[PeriodicProcessDeploymentStatus]]).
+   * For multiple schedules only the effective one is returned.
+   */
   def getLatestDeployment(processName: ProcessName): Future[Option[PeriodicProcessDeployment]] = {
```

Comment:
Is it enough to handle e.g. 'deactivate' method properly?

Reply:
I think it should be fine. In deactivate method, latest deployment is passed to listener and in case of a failed job, it could be passed to the listener instead of newer deployment in current behaviour... Maybe we should pass to listeners all latest deployments for each schedule.